modules/lib/products.js

```diff
@@ -34,3 +36,32 @@ module.exports.CLIENTS = {
     groups: getSpecificGroupIds(['firstorderteam', 'snoketeam'])
   },
 }
+
+module.exports.getSetLock = (dbName) => async ({ user_id, channel, product }) => {
+  if (!process.env.DETA_KEY) {
+    return true
+  }
+  const deta = Deta(process.env.DETA_KEY)
+  const db = deta.Base(dbName)
+  const key = `${channel}::${product}`
+  const exists = await db.get(key)
```

Comment:
If you don't care about the return payload, you can do this operation in one go using the `insert` method which will throw an error if the key already exists. Might also prevent any insertion between the `get` and `put` calls where a competing lambda could acquire the lock (although I would assume the probability of this to be extremely low given our low traffic use case).

Reply:
Also, perhaps there should be some sort of TTL on the key so that it doesn't block the application permanently if not released. This could be implemented via the key itself (e.g. `${channel}::${product}::${minute for which it is valid}`). What do you think?

Reply:
that's a good call to use `::{minutely, or some sort of period}` to compose the key, until we get official TTL support from deta directly